### DISCUSSION TITLE
Use file for configure meson.build

### DISCUSSION THREAD

**Discussion by vbextreme** [mesonbuild/meson#13259] [Q&A] [answered]
I wanted to know if it is possible to create a file to configure Meson, that is, without having to pass -D from the command line, but to directly pass a file with all the configurations inside.
My project has about 20 subprojects and around 50+ configurations. Passing them from the command line every time is impractical. Currently, I do it like this:
```
cat config/arch.ini | xargs meson setup build-arch
```
but I find it unsafe and inelegant
thanks
have a good life

**Comment by eli-schwartz** [comment, marked as answer]
https://mesonbuild.com/Machine-files.html

They are ini files and can look like this:

```
[built-in options]
buildtype = 'debugoptimized'

[project options]
build-tests = true

[zlib:project options]
build-tests = false

[othersubproject:project options]
build-tests = false
```

**Comment by vbextreme** [comment]
Thanks
